### DISCUSSION TITLE
Shipping service not automatically set on cart

### DISCUSSION THREAD

**Discussion by Kealtie** [swellstores/community#456] [Help]
Hi,

I've noticed on my project that the standard shipping rule is not automatically set when a new cart is created, even though the cart complies with all the set rules on the shipping service. How can I make it so that the shipping service is automatically applied to a newly created cart?

![image](https://github.com/swellstores/community/assets/7847482/247ea925-4843-4a9b-9f9b-613c0fe3df18)

Greetings,
Jordy

**Comment by ericingram** [comment]
Can you detail the steps you're taking? 

Shipment services are typically not applied without a specific action. Checkout will usually make an API call to get rates and then another call when the user selects a service. From the dashboard, I believe you have to select one manually if it wasn't set by an API call or something like that already.

**Reply by Kealtie** [reply]
I'm just adding a product to my cart, then moving to my cart. Which in our project also displays the shipping costs at that point.

But what I understand from your perspective is that a shipping service is not automatically applied and also not possible to work this way?

**Reply by Kealtie** [reply]
Also, if the above is true, how would I add a shipping service to a cart without the interception from a back-end API. But straight from the front-end API?

**Reply by ericingram** [reply]
Do you mean moving to the cart in checkout or somewhere else?

This page in the docs shows how to get shipping rates in the cart from the frontend API: https://developers.swell.is/frontend-api/carts#shipping-rates

**Reply by logeshswell** [reply]
The below doc helps you to set the shipping service via swell-js:
https://developers.swell.is/frontend-api/carts#applying-shipping-rate-to-cart

**Reply by Kealtie** [reply]
Thanks for the replies @logeshswell & @ericingram 

I've added the shipping service to the cart by using the above mentioned endpoint. Like this
`await swell.cart.update({
            "shipping": {
                "service": 'basic'
            }
        });`

This does at the shipping service to the cart, but does not apply the shipping price specified in the service to the cart.

Also I've tried adding `await swell.cart.getShippingRates();` after the addition of the service to the cart, but this does not change the outcome.

Why does the price specified in the given shipping service not get applied to the cart?

![image](https://github.com/swellstores/community/assets/7847482/5c168a53-2f08-4122-af03-5d6c02e3a6b1)

**Reply by logeshswell** [reply]
Did you try fetching the shipping rates first followed by updating the shipping service?

**Reply by Kealtie** [reply]
Do you mean like this?

![image](https://github.com/swellstores/community/assets/7847482/ac85b4f1-1f6b-40f2-8b55-318800ffcb09)

Wouldn't I have to set the right shipping service first?

This didn't do the trick. Is it possible to schedule a call of some sort to look at this problem together?

**Reply by logeshswell** [reply]
Now, can you confirm whether your problem is, the shipping price is not applied correctly in the cart?
If so, what's the response do you find in `getShippingRates()`?

**Reply by Kealtie** [reply]
<img width="667" alt="image" src="https://github.com/swellstores/community/assets/7847482/bd14457e-c05c-40c3-8838-67e46acaf94d">

Yes, I have applied the 'Basic' shipping service which should apply 97,50 euro's in shipping costs. The response I get from `  await swell.cart.getShippingRates();`, is my cart, without the right shipping price applied.

Is it possible to schedule a call?

**Reply by logeshswell** [reply]
Let me check and update you shortly now.

**Reply by logeshswell** [reply]
The issue is shipment_rating is set as `null` and it could be because you didn't set `shipping.country` value.
<img width="798" alt="Screenshot 2023-12-27 at 7 34 52 PM" src="https://github.com/swellstores/community/assets/100766131/444a0989-192f-45c3-86ca-c2dcfd5b8d27">

Can you try setting the shipping country before fetching shipment rates, so that shipment_rating should contain a list of shipping services?

**Reply by Kealtie** [reply]
Alright, I've applied the following Shipping zone 'Worldwide', selecting 'All remaining area's'. Which I assume then applies it to everywhere in the world.

![image](https://github.com/swellstores/community/assets/7847482/9d84f6cc-a832-4edf-83ed-7ea5de30f656)

Problem in my specific use case is I want to display the shipping costs in the 'shopping cart', before a potential customer actually fills in their shipping information. 

Currently both applying the 'basic' shipping service and fetching the shipping rates does not set the right shipping price on the cart. Shipping price retrieved remains 0,00, which should be 97,50. Also i've tried adding rules to the shipping service that state if the order value of the cart is above 1450 euro's there are no shipping costs, lower then 1450 would apply 97,50 shipping costs.

**Reply by logeshswell** [reply]
Still, you need to set `shipping.country` to some default value, so that `shipment_rating` will be computed and default shipping cost will be applied.

**Reply by Kealtie** [reply]
Ah yes, I figured. Thanks.

But in my particular use case I want to show the shipping costs (97,50) to the customer before checkout, so, I do not know their address information yet. How would I implement this?

**Reply by logeshswell** [reply]
Since the shipping cost doesn't change w.r.t country, set the default country as the country in which the application is being accessed (just for an instance).
